astropy/utils/console.py

```diff
@@ -631,10 +637,17 @@ def _update_ipython_widget(self, value=None):
         # if none exists.
         if not hasattr(self, '_widget'):
             # Import only if an IPython widget, i.e., widget in iPython NB
-            from IPython.html import widgets
+            try:
+                from ipywidgets import widgets
+            except ImportError:
+                from IPython.html import widgets
             from IPython.display import display
 
-            self._widget = widgets.FloatProgressWidget()
+            try:
+                self._widget = widgets.FloatProgress()
+            except AttributeError:
+                self._widget = widgets.FloatProgressWidget()
```

Comment:
Thanks, I've updated this now to use `ipywidgets` and `FloatProgress` only for `ipython >= 4`.